From 0fb3a5437c87c789cece2c59c6d9381155343b50 Mon Sep 17 00:00:00 2001
From: Leith Bade <leith@bade.nz>
Date: Thu, 27 Jun 2024 17:58:55 +1000
Subject: [PATCH 3/4] arm64: dts: mediatek: mt7986: add missing pin groups to
 BPI-R3

The uart0, mdio and switch nodes are not connected to pin groups unlike
most of the other nodes.

Signed-off-by: Leith Bade <leith@bade.nz>
---
 .../dts/mediatek/mt7986a-bananapi-bpi-r3.dts  | 27 +++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts b/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts
index 61a23ab35ea3..044b9e4534f7 100644
--- a/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts
+++ b/arch/arm64/boot/dts/mediatek/mt7986a-bananapi-bpi-r3.dts
@@ -190,6 +190,8 @@ gmac1: mac@1 {
 	};
 
 	mdio: mdio-bus {
+		pinctrl-names = "default";
+		pinctrl-0 = <&mdio_pins>;
 		#address-cells = <1>;
 		#size-cells = <0>;
 	};
@@ -203,6 +205,8 @@ switch: switch@31 {
 		interrupt-parent = <&pio>;
 		interrupts = <66 IRQ_TYPE_LEVEL_HIGH>;
 		#interrupt-cells = <1>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&switch_pins>;
 		reset-gpios = <&pio 32 GPIO_ACTIVE_HIGH>;
 	};
 };
@@ -332,6 +336,13 @@ mux {
 		};
 	};
 
+	uart0_pins: uart0-pins {
+		mux {
+			function = "uart";
+			groups = "uart0";
+		};
+	};
+
 	uart1_pins: uart1-pins {
 		mux {
 			function = "uart";
@@ -346,6 +357,20 @@ mux {
 		};
 	};
 
+	mdio_pins: mdio-pins {
+		mux {
+			function = "eth";
+			groups = "mdc_mdio";
+		};
+	};
+
+	switch_pins: switch-pins {
+		mux {
+			function = "eth";
+			groups = "switch_int";
+		};
+	};
+
 	wf_2g_5g_pins: wf-2g-5g-pins {
 		mux {
 			function = "wifi";
@@ -468,6 +493,8 @@ &trng {
 };
 
 &uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_pins>;
 	status = "okay";
 };
 
-- 
2.51.2

